addon/components/frost-modal-binding.js

```diff
@@ -134,6 +138,38 @@ const FrostModalBinding = Component.extend(PropTypesMixin, {
 
   // == Actions ===============================================================
 
+  /**
+   * Disable the Confirm button and defer calling onClose and until after onConfirm() resolves
+   * @param {Promise} confirmed - the result of calling onConfirm
+   * @returns {Promise} a promise chain with errors shal
+   * @private
+   */
+  _handleThenableOnConfirmResult (confirmed) {
+    // handle closeOnConfirm + button-disabling separately, so that we still protect from spamming
+    // even if we want to avoid automatic invocation of onClose
+
+    if (this.get('closeOnConfirm') === true) {
+      confirmed = confirmed.then(() => {
+        this.onClose()
+      })
+    }
+
+    confirmed = confirmed.catch((err) => {
+      deps.Logger.log(err)
```

Comment:
Do we not want it in the log as an error?

Reply:
We do, thanks for catching that! I don't think I have ever used `Logger.error` before, so I forgot it existed.